test/test_expression.cpp

```diff
@@ -76,7 +76,7 @@ TEST_CASE("Expression")
         REQUIRE(exp_t::get_precedence(FUN_CALL) >= exp_t::get_precedence(ARRAY));
         REQUIRE(exp_t::get_precedence(ARRAY) >= exp_t::get_precedence(DOT));
         REQUIRE(exp_t::get_precedence(DOT) >= exp_t::get_precedence(RATE));
-        REQUIRE(exp_t::get_precedence(RATE) > exp_t::get_precedence(PRE_INCREMENT));
+        REQUIRE(exp_t::get_precedence(PRE_INCREMENT) > exp_t::get_precedence(ARRAY));
```

Comment:
since the precedence has changed, please move this line lower, to keep the precedence monotonically decreasing
```suggestion
        REQUIRE(exp_t::get_precedence(PRE_INCREMENT) >= exp_t::get_precedence(NOT));
        REQUIRE(exp_t::get_precedence(NOT) > exp_t::get_precedence(ARRAY));
```

Reply:
Actually, something is wrong with this precedence, array subscripting is supposed to have higher precedence than pre-increment:
https://en.cppreference.com/w/c/language/operator_precedence

Reply:
Good catch, I misunderstood the code. I thought low numbers where evaluated first. I've added a comment to clarify